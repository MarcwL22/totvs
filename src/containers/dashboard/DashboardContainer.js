import React, { Component, Fragment } from 'react';
import axios from 'axios';
import styled from 'styled-components';
// UIComponents
import { CenterContent, Loading } from '../../components';
// Components
import DashboardModal from './components/DashboardModal';
import DashboardList from './components/DashboardList';
import DashboardResultInfo from './components/DashboardResultInfo';
import DashboardHeader from './components/DashboardHeader';

const DashboardSectionContent = styled.div`
  min-height: 70vh;
  position: relative;
`;

export class DashboardContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchInput: '',
      loading: false,
      repositoriesList: [],
      error: false,
      modalOpen: false
    };
  }

  onChangeText = e => {
    const { value } = e.target;
    this.setState({ searchInput: value });
  };

  openModal = () => this.setState({ modalOpen: true });

  searchFormSubmit = e => {
    e.preventDefault();
    const { searchInput } = this.state;
    this.setState({ loading: true }, () => {
      axios
        .get(`https://api.github.com/users/${searchInput}/repos`)
        .then(({ data }) => this.setState({ loading: false, repositoriesList: data, error: false }))
        .catch(() => this.setState({ loading: false, error: true, repositoriesList: [] }));
    });
  };

  renderContent = () => {
    const { loading, repositoriesList, error } = this.state;
    const Content = () => {
      if (loading) {
        return <Loading />;
      } else if (!loading && error) {
        return <DashboardResultInfo type="error" />;
      } else {
        return <DashboardResultInfo />;
      }
    };
    if (repositoriesList.length === 0) {
      return <CenterContent>{Content()}</CenterContent>;
    }
    return <DashboardList data={repositoriesList} openModal={this.openModal} />;
  };

  render() {
    const { searchInput, modalOpen } = this.state;
    return (
      <Fragment>
        <DashboardHeader onChangeText={this.onChangeText} value={searchInput} onSubmit={this.searchFormSubmit} />
        <main>
          <DashboardSectionContent>{this.renderContent()}</DashboardSectionContent>
        </main>
        <DashboardModal modalOpen={modalOpen} closeModal={() => this.setState({ modalOpen: false })} />
      </Fragment>
    );
  }
}

export default DashboardContainer;
