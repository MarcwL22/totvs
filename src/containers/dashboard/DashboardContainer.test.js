import React from 'react';
import { shallow } from 'enzyme';
import { DashboardContainer } from './DashboardContainer';

describe('DashboardContainer', () => {
  let dashboard = shallow(<DashboardContainer />);
  it('should renders properly', () => {
    expect(dashboard).toMatchSnapshot();
  });
});
