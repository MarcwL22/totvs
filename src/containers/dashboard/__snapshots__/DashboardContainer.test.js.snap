// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardContainer should renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashboardContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <DashboardHeader
          onChangeText={[Function]}
          onSubmit={[Function]}
          value=""
/>,
        <main>
          <styled.div>
                    <styled.div>
                              <DashboardResultInfo
                                        type="empty"
                              />
                    </styled.div>
          </styled.div>
</main>,
        <DashboardModal
          closeModal={[Function]}
          modalOpen={false}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChangeText": [Function],
          "onSubmit": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <styled.div>
            <styled.div>
                        <DashboardResultInfo
                                    type="empty"
                        />
            </styled.div>
</styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.div>
              <DashboardResultInfo
                            type="empty"
              />
</styled.div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <DashboardResultInfo
                type="empty"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "type": "empty",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "closeModal": [Function],
          "modalOpen": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <DashboardHeader
            onChangeText={[Function]}
            onSubmit={[Function]}
            value=""
/>,
          <main>
            <styled.div>
                        <styled.div>
                                    <DashboardResultInfo
                                                type="empty"
                                    />
                        </styled.div>
            </styled.div>
</main>,
          <DashboardModal
            closeModal={[Function]}
            modalOpen={false}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChangeText": [Function],
            "onSubmit": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <styled.div>
              <styled.div>
                            <DashboardResultInfo
                                          type="empty"
                            />
              </styled.div>
</styled.div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.div>
                <DashboardResultInfo
                                type="empty"
                />
</styled.div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <DashboardResultInfo
                  type="empty"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "type": "empty",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "closeModal": [Function],
            "modalOpen": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
